docs/guides/wifi.md

```diff
@@ -2,27 +2,90 @@
+that shipped with Catalina never had Mojave firmware made, so this
+patchset does not support wifi on those models (for now). It currently
+supports the BCM4364 and BCM4355 chipsets.
+
+### Corellium's Patchset
+
+[This patch](https://github.com/corellium/linux-m1/commit/02ad06fbf2b35916ee329a9bb80d73840d6e2973)
+was made by Corellium for M1 Macs and uses wifi firmware in Big Sur's
+format. It is the default for mbp-fedora and mbp-nixos. Kernels with
+this patch are available for [Ubuntu/Debian](https://github.com/Redecorating/mbp-ubuntu-kernel/releases)
+and [Arch based distros](https://github.com/Redecorating/mbp-16.1-linux-wifi/releases),
+but you can [compile it yourself](#compiling-with-corelliums-patchset)
+if you need/want to. This supports the BCM4364 Chipset, and with
+additional patches [should support the BCM4377 Chipset](#BCM4377).
+
+#### Compiling with Corellium's Patchset
+
+Follow the [kernel compiling guide](https://wiki.t2linux.org/guides/kernel/#compile),
+But make sure to use [https://github.com/jamlam/mbp-16.1-linux-wifi](https://github.com/jamlam/mbp-16.1-linux-wifi)
+instead of [https://github.com/aunali1/linux-mbp-arch](https://github.com/aunali1/linux-mbp-arch)
+as the patchset repository.
+
+#### BCM4377
+
+The BCM4377 chipset is Broadcom's "2.0" hardware, and requires changes to
+the `brcmfmac` driver. The artifact from [this CI run](https://github.com/Redecorating/mbp-16.1-linux-wifi/actions/runs/1037316726)
+has an Arch kernel with extra patches in addition to the
+Corellium patch that should support the BCM4377. If you test this, you
+will need Big Sur wifi firmware.
 
 ## On macOS
 
 1. Run both `ioreg -l | grep RequestedFiles` (the names of the firmware files required by your model) and `system_profiler SPHardwareDataType | grep "Model Identifier"` (your model identifier) in a terminal and note down the output, you will need both values in the next steps.
 2. If your model identifier is 16,1 you may get the firmware files from [https://github.com/AdityaGarg8/mbp-16.1-wifi-firmware](https://github.com/AdityaGarg8/mbp-16.1-wifi-firmware) by following the instructions given there. On getting the correct files navigate to [On Linux](https://wiki.t2linux.org/guides/wifi/#on-linux) section. If you unable to get working firmware files from there then follow the instructions given for other models.
 3. For other models go to [On any OS](https://wiki.t2linux.org/guides/wifi/#on-any-os) section.
 
-> Note: If you are unable to get the firmware even after following guide for other models, you might need to extract firmware from macOS at this point. See the instructions in [On any OS](https://wiki.t2linux.org/guides/wifi/#on-any-os) section, especially the first note there.
+> Note: If you are unable to get the firmware even after following guide for other models, you might need to extract firmware from macOS at this point. See the instructions in [On any OS](https://wiki.t2linux.org/guides/wifi/#on-any-os) section.
```

Comment:
Maybe this could actually be merged with the table. In that case "Is my model supported" might better be "Getting the right firmware" or something